lib/parser/prescan.cc

```diff
@@ -545,7 +545,7 @@ void Prescanner::QuotedCharacterLiteral(
   bool escape{false};
   bool escapesEnabled{features_.IsEnabled(LanguageFeature::BackslashEscapes)};
   while (true) {
-    char ch{*at_};
+    unsigned char ch{static_cast<unsigned char>(*at_)};
```

Comment:
`char32_t` might be better since that's what `EmitQuotedChar`'s argument is declared as.  But did this change fix anything?

Reply:
Yes `char32_t` is more appropriate for `ch`, I have changed that.
The cast to `unsigned char` changed something (at least on the machine I was testing were `char` is signed). Take "你" for instance (U+20320), its UTF-8 encoding is` 0xE4 0xBD 0xA0`. Without the cast, the first byte `0XE4` ended up interpreted as `-28` as a `char`, that was casted to `4294967267` as a `char32_t` in `EmitQuotedChar` instead of `228`. As a result, `EmitQuotedChar` was emitting `"\367\277\277\244"` instead of `"\344"` and "你" was pre-scanned as `"\367\277\277\244\367\277\276\275\367\277\276\240"` instead of `"\344\275\240"`